sdk/src/contexts/solana/utils/utils/transaction.ts

```diff
@@ -206,3 +142,71 @@ export async function sendAndConfirmTransactionsWithRetry(
 
   return Promise.resolve(output);
 }
+
+// This function signs and sends the transaction while constantly checking for confirmation
+// and resending the transaction if it hasn't been confirmed after the specified interval
+// NOTE: The caller is responsible for simulating the transaction and setting any compute budget
+// or priority fee before calling this function
+// See https://docs.triton.one/chains/solana/sending-txs for more information
+export async function signSendAndConfirmTransaction(
+  connection: Connection,
+  signTransaction: SignTransaction,
+  payer: PublicKeyInitData,
+  unsignedTransaction: Transaction,
+  commitment: Commitment = 'finalized',
+  txRetryInterval = 5000,
+): Promise<TransactionSignatureAndResponse> {
+  const { blockhash, lastValidBlockHeight } =
+    await connection.getLatestBlockhash({
+      commitment,
+    });
+  unsignedTransaction.recentBlockhash = blockhash;
+  unsignedTransaction.feePayer = new PublicKey(payer);
+  const tx = await signTransaction(unsignedTransaction);
+  let confirmTransactionPromise: Promise<
+    RpcResponseAndContext<SignatureResult>
+  > | null = null;
+  let confirmedTx: RpcResponseAndContext<SignatureResult> | null = null;
+  let txSendAttempts = 1;
+  let signature = '';
+  const serializedTx = tx.serialize();
+  const sendOptions = {
+    skipPreflight: true,
+    maxRetries: 0,
+    preFlightCommitment: commitment, // See PR and linked issue for why setting this matters: https://github.com/anza-xyz/agave/pull/483
+  };
+  signature = await connection.sendRawTransaction(serializedTx, sendOptions);
+  confirmTransactionPromise = connection.confirmTransaction(
+    {
+      signature,
+      blockhash,
+      lastValidBlockHeight,
+    },
+    commitment,
+  );
+  // Keep checking for confirmation until the promise resolves or the timeout is reached
+  while (!confirmedTx) {
+    confirmedTx = await Promise.race([
+      confirmTransactionPromise,
+      new Promise<null>((resolve) =>
+        setTimeout(() => {
+          resolve(null);
+        }, txRetryInterval),
+      ),
+    ]);
+    if (confirmedTx) {
+      break;
+    }
+    console.log(
+      `Tx not confirmed after ${
+        txRetryInterval * txSendAttempts++
+      }ms, resending`,
+    );
+    // Resend the transaction if it hasn't been confirmed
+    await connection.sendRawTransaction(serializedTx, sendOptions);
```

Comment:
don't you need to get a fresh blockhash before doing this?

Reply:
I don't think it matters since we definitely don't want to change the `lastValidBlockHeight` - we want to rebroadcast the same signed TX until it's confirmed or not